### PR TITLE
Process debugger auto revert log level

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- the process debugger output a lot of log data
  
## Proposed Changes
- made a new configurable option so the processDebugger can shut itself down and revert the log level after a defined time. Right now this time interval is set to 10 minutes

## Testing procedure
- standard testing procedure. If we see the `node is stuck` WARN message, the TRACE logs will be available for ~10 minutes

### CODE REVIEW

File: debug/process/debugger.go [hunk 1]
```diff
@@ -158,10 +160,31 @@ func dumpGoRoutines() {
 }
 
 func (debugger *processDebugger) changeLog() {
+	oldLogLevel := logger.GetLogLevelPattern()
+
 	errSetLogLevel := logger.SetLogLevel(debugger.debuggingLogLevel)
 	if errSetLogLevel != nil {
-		log.Error("debugger.changeLog: cannot change log level", "error", errSetLogLevel)
+		log.Error("processDebugger: cannot change log level", "error", errSetLogLevel)
+	}
```
Comment (SebastianMarian) [inline debug/process/debugger.go:168]: Missing return here?

File: debug/process/debugger.go [hunk 2]
```diff
@@ -158,10 +160,31 @@ func dumpGoRoutines() {
...
+func (debugger *processDebugger) revertLogLevel(oldLogLevel string) {
+	timeToWait := time.Second * time.Duration(debugger.revertTimeInSeconds)
+	log.Debug("processDebugger revertLogLevel", "original log level", oldLogLevel, "will revert in", timeToWait)
+
+	time.Sleep(timeToWait)
+
+	errSetLogLevel := logger.SetLogLevel(oldLogLevel)
+	if errSetLogLevel != nil {
+		log.Error("processDebugger changeLog: cannot change log level", "error", errSetLogLevel)
 	}
```
Comment (SebastianMarian) [inline debug/process/debugger.go:186]: Missing return here?

File: debug/process/debugger.go [hunk 3]
```diff
@@ -158,10 +160,31 @@ func dumpGoRoutines() {
...
+
+func (debugger *processDebugger) revertLogLevel(oldLogLevel string) {
+	timeToWait := time.Second * time.Duration(debugger.revertTimeInSeconds)
+	log.Debug("processDebugger revertLogLevel", "original log level", oldLogLevel, "will revert in", timeToWait)
+
+	time.Sleep(timeToWait)
+
+	errSetLogLevel := logger.SetLogLevel(oldLogLevel)
+	if errSetLogLevel != nil {
+		log.Error("processDebugger changeLog: cannot change log level", "error", errSetLogLevel)
```
Comment (SebastianMarian) [inline debug/process/debugger.go:185]: changeLog or revertLogLevel? changeLog is the name of the above method

File: cmd/node/config/config.toml [hunk 4]
```diff
@@ -820,6 +820,8 @@
         DebuggingLogLevel = "*:DEBUG,p2p:TRACE,debug:DEBUG,process:TRACE,intercept:TRACE"
         GoRoutinesDump = true
         PollingTimeInSeconds = 240 # 4 minutes
+        # setting this to 0 disbles the automatic revert of the log level
```
Comment (sstanculeanu) [inline cmd/node/config/config.toml:823]:
```suggestion
        # setting this to 0 disables the automatic revert of the log level
```

File: debug/process/debugger.go (hunk 1 above)
Comment (iulianpascalau) [reply debug/process/debugger.go:168]: yes, we might return here as to retry the log change in about 4minutes

File: debug/process/debugger.go (hunk 2 above)
Comment (iulianpascalau) [reply debug/process/debugger.go:186]: added

File: debug/process/debugger.go (hunk 3 above)
Comment (iulianpascalau) [reply debug/process/debugger.go:185]: right, too much copy paste

File: cmd/node/config/config.toml (hunk 4 above)
Comment (iulianpascalau) [reply cmd/node/config/config.toml:823]: fixed

**Review by SebastianMarian — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**